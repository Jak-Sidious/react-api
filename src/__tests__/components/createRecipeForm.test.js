import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import { shallowToJson} from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';

import CreateRecipeForm from '../../components/commonComponents/createRecipeForm';

Enzyme.configure({ adapter: new Adapter() });

describe('<CreateRecipeForm />', () => {
  const pathname = 'path';
  const props = {
    post: jest.fn(() => Promise.resolve('landing')),
    onChange: jest.fn(),
    handleChange: jest.fn(),
    history: { push: jest.fn() },
    location: { pathname },
    match: {
      param: {
        id: 1
      }
    }
  };
  const preventDefault = jest.fn();
  const component = mount(<MemoryRouter><CreateRecipeForm {...props} /></MemoryRouter>)
  it('should render without crashing', () => {
    const { enzymeWrapper } = mount(
      <MemoryRouter>
        <CreateRecipeForm {...props} />
      </MemoryRouter>
    );
  });
  it('should render properly', () => {
    const wrapper = shallow(<CreateRecipeForm {...props}/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render form', () => {
    expect(component.find('Form').length).toBe(1);
    expect(component.find('Form').simulate("submit", { preventDefault }));
    expect(preventDefault).toBeCalled();

  })
});
