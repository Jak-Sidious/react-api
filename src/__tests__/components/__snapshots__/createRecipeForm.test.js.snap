// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRecipeForm /> should render properly 1`] = `
<div>
  <Navigation
    pathname="path"
  />
  <_class
    options={Object {}}
  />
  <div
    className="RecipeCreate"
  >
    <div
      className="wrapper"
    >
      <Form
        as="form"
        className="form-signin"
      >
        <h2
          className="form-signin-heading"
        >
          Create Recipe Form
        </h2>
        <FormField
          className="categoryName"
          control={[Function]}
          fluid={true}
          name="categoryName"
          onChange={[Function]}
          options={Array []}
          placeholder="Select Category"
          selection={true}
          type="categoryName"
        />
        <FormInput
          as={[Function]}
          autoFocus={true}
          className="recName"
          control={[Function]}
          fluid={true}
          name="RecipeName"
          onChange={[Function]}
          placeholder="Enter the Recipe Name"
          type="RecipeName"
        />
        <FormTextArea
          as={[Function]}
          autoFocus={true}
          className="ingrid"
          control={[Function]}
          name="ingredients"
          onChange={[Function]}
          placeholder="Enter your ingredients"
          type="ingredients"
        />
      </Form>
    </div>
  </div>
</div>
`;
